From: Anders Kaseorg <andersk@mit.edu>
Date: Tue, 14 Jun 2011 19:24:42 +0000 (-0400)
Subject: Treat Linux 3 as Linux 2.6
X-Git-Url: 
http://git.openafs.org/?p=openafs.git;a=commitdiff_plain;h=34bdd979fc487843c26f15655b94eac7c8f98fee;hp=cdd9f78f058f6d6c67d52d8b28a66f24065702d9

Treat Linux 3 as Linux 2.6

Linux 3.0 is just a rebranded Linux 2.6.40, so we can use the same
sysname and the same code.

Change-Id: I154334dc5b73980a0d9b8abd9fd6645994f63829
Signed-off-by: Anders Kaseorg <andersk@mit.edu>
Reviewed-on: http://gerrit.openafs.org/4843
Reviewed-by: Derrick Brashear <shadow@dementia.org>
Reviewed-by: Simon Wilkinson <sxw@inf.ed.ac.uk>
Tested-by: BuildBot <buildbot@rampaginggeek.com>
---

diff --git a/acinclude.m4 b/acinclude.m4
index 5fd2c29..dd2a137 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -358,10 +358,12 @@ case $system in
 		fi
 		AC_MSG_RESULT(linux)
                 if test "x$enable_kernel_module" = "xyes"; then
-                 AFS_SYSKVERS=`echo $LINUX_VERSION | awk -F\. '{print $[]1 $[]2}'`
-                 if test "x${AFS_SYSKVERS}" = "x"; then
-                  AC_MSG_ERROR(Couldn't guess your Linux version [2])
-                 fi
+                 case "$LINUX_VERSION" in
+                  2.2.*) AFS_SYSKVERS=22 ;;
+                  2.4.*) AFS_SYSKVERS=24 ;;
+                  2.6.* | 3.*) AFS_SYSKVERS=26 ;;
+                  *) AC_MSG_ERROR(Couldn't guess your Linux version [2]) ;;
+                 esac
                 fi
                 ;;
         *-solaris*)
